Use SPIFFE in TLS mode

Is it possible to use SPIFFE as source for the CA bundle in client mode but with TLS only? I tried it with `--disable-authentication` but it fails. I also didn't found another solution then manually fetching the CA bundle. The proxy is not started until the client agent responds with an SVID.

Using `--use-workload-api-addr` with `--allow-all` on the server mode also doesn't work. Seems like the only way to get TLS-only mode with SVIDs is to use the spiffe-helper, right? This is sadly not possible using Windows, since Ghostunnel does not support refresh signals on Windows. So there is no real way to get certificate rotation done unless Ghostunnel is supporting TLS-only mode directly as far as I can see it.

I was not able to build the Windows version with the Makefile using Docker.

```
ghostunnel-master>docker.exe run -it --rm -v ghostunnel-master:/usr/src/app -e GOOS=windows -e GOARCH=386 golang:1.18
root@8fd865132a41:/go# cd /usr/src/app
root@8fd865132a41:/usr/src/app# make
go build -ldflags '-X main.version=cd0886b' -o ghostunnel .
# github.com/ghostunnel/ghostunnel/certstore
certstore/certstore.go:22:9: undefined: openStore
make: *** [Makefile:7: ghostunnel] Error 2
```

--- a/certloader/spiffe_tls_config.go
+++ b/certloader/spiffe_tls_config.go
@@ -110,7 +110,12 @@ func (c *spiffeTLSConfig) GetClientConfig() *tls.Config {
 
 func (c *spiffeTLSConfig) GetServerConfig() *tls.Config {
 	config := c.base.Clone()
-	config.ClientAuth = tls.RequireAnyClientCert
+
+	// Only set client requirement if not disabled in base.
+	if config.ClientAuth == tls.RequireAndVerifyClientCert {
+		config.ClientAuth = tls.RequireAnyClientCert
+	}
+
 	// Go TLS stack will do hostname validation with is not a part of SPIFFE
 	// authentication. Unfortunately there is no way to just skip hostname
 	// validation without having to turn off all verification. This is still

--- a/main.go
+++ b/main.go
@@ -351,13 +351,11 @@ func clientValidateFlags() error {
 		(*certPath != "" && *keyPath != ""),
 		// A certificate, with the key in a PKCS#11 module
 		(*certPath != "" && hasPKCS11()),
-		// SPIFFE Workload API
-		*useWorkloadAPI,
 		// No credentials needed if auth is disabled
 		*clientDisableAuth,
 	})
 
-	if hasValidCredentials == 0 {
+	if hasValidCredentials == 0 && !*useWorkloadAPI {
 		return errors.New("at least one of --keystore, --cert/--key, --keychain-identity/issuer (if supported) or --disable-authentication flags is required")
 	}
 	if hasValidCredentials > 1 {
@@ -577,7 +575,7 @@ func serverListen(context *Context) error {
 	}
 
 	if *serverDisableAuth {
-		config.ClientAuth = tls.NoClientCert
+		config.ClientAuth = tls.RequestClientCert
 	} else {
 		config.VerifyPeerCertificate = serverACL.VerifyPeerCertificateServer
 	}

